tests/e2e/performance/csi_tests/test_pvc_multi_snapshot_performance.py

```diff
@@ -1,42 +1,445 @@
+
+        res = run_oc_command(cmd=f"create -f {tmpfile}", namespace=self.nss_name)
+        if ERRMSG in res[0]:
+            err_msg = f"Failed to create snapshot : {res}"
+            log.error(err_msg)
+            raise Exception(err_msg)
+
+        # wait until snapshot is ready
+        timeout = 600
+        sleep_time = 5
+
+        while timeout > 0:
+            res = run_oc_command(
+                f"get volumesnapshot {snap_name} -o yaml", namespace=self.nss_name
+            )
+
+            if ERRMSG not in res[0]:
+                res = yaml.safe_load("\n".join(res))
+                log.debug(f"The command output is : {yaml.dump(res)}")
+                try:
+                    if res["status"]["readyToUse"]:
+                        log.info(f"{snap_name} Created and ready to use")
+                        snap_con_name = res["status"]["boundVolumeSnapshotContentName"]
+                        break
+                    else:
+                        log.info(
+                            f"{snap_name} is not ready yet, sleep {sleep_time} sec before re-check"
+                        )
+                        time.sleep(sleep_time)
+                        timeout -= sleep_time
+                except Exception:
+                    log.info(
+                        f"{snap_name} is not ready yet, sleep {sleep_time} sec before re-check"
+                    )
+                    time.sleep(sleep_time)
+                    timeout -= sleep_time
+
+            else:
+                err_msg = f"Can not get snapshot status {res}"
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+        return self.get_creation_time(snap_name, snap_con_name, UTC_datetime)
+
+    def read_logs(self, kind, namespace, start_time):
+        """
+        Reading the csi-driver logs, since we use different logs for the start time
+        for end time (creation snapshot), we call this function twice.
+
+        Args:
+            kind (str): the kind of logs to read 'start' or 'end'
+            namespace (str): in which namespace the pod exists
+            start_time (time): the start time of the specific test,
+               so we dont need to read the full log
+
+        Returns:
+            list : the contant of all read logs(s) - can be more then one log
+
+        """
+        logs = []
+        # The pod with the logs for 'start' creation time have only one container
+        container = ""
+        if kind == "end":
+            # The pod with the logs for 'end' creation time have more then one container
+            container = "-c csi-snapshotter"
+        for l in self.log_names[kind]:
+            logs.append(
+                run_oc_command(
+                    f"logs {l} {container} --since-time={start_time}",
+                    namespace=namespace,
+                )
+            )
+        return logs
+
+    def get_creation_time(self, snap_name, content_name, start_time):
```

Comment:
same here, good to move it to common library

Reply:
same as above.